src/jit/morph.cpp

```diff
@@ -3312,15 +3303,58 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
 #else // FEATURE_UNIX_AMD64_STRUCT_PASSING
                         if (!structDesc.passedInRegisters)
                         {
+                            GenTreePtr lclVar = fgIsIndirOfAddrOfLocal(argObj);
+                            bool needCpyBlk = false;
+                            if (lclVar != NULL)
+                            {
+                                // If the struct is promoted to registers, it has to be materialized
+                                // on stack. We may want to support promoted structures in
+                                // codegening pugarg_stk instead of creating a copy here.
+                                LclVarDsc*  varDsc = &lvaTable[lclVar->gtLclVarCommon.gtLclNum];
+                                needCpyBlk = varDsc->lvPromoted;
+                            }
+                            else 
+                            {
+                                // The following pattern needs copy to store the
+                                // result of the simd operation. Otherwise, this                               
+                                // pattern reaches codegen that doesn't expect
+                                // addr node.
```

Comment:
I would suggest to rephrase it as

"GT_ADDR(GT_SIMD) is not a rationalized IR form and is not handled by rationalizer.  For now we will let SIMD struct arg to be copied to a local.  As part of cpblk rewrite, rationalizer will handle GT_ADDR(GT_SIMD)".

I would also suggest to add a TODO and open a git issue to track this.